types/index.d.ts

```diff
@@ -2,15 +2,15 @@ import {ChartType, Plugin} from 'chart.js';
 import {Options} from './options';
 
 declare module 'chart.js' {
-  interface ChartDatasetProperties<TType extends ChartType, TData extends unknown[]> {
+  interface ChartDatasetProperties<TType extends ChartType, TData> {
```

Comment:
This is probably an issue with the types in beta.12

Reply:
If the generic base type is still needed, maybe the core could export an nicer alias:

```ts
// chart.js
export type AnyArray = unknown[];
```

```ts
// chart-plugin-datalabels
import { AnyArray } from 'chart.js';

declare module 'chart.js' {
  interface ChartDatasetProperties<TType extends ChartType, TData extends AnyArray> {
    //...
}
```

Reply:
Could even be `AnyData` instead of `AnyArray`.

Reply:
I am just matching the definition here to prevent errors. If only array data-types are needed, would `TData extends Array<any>` work?

Reply:
I think it needs to be the exact same definition as in the core repo, so it would not work. Also, I disabled the use of `any` and enforce array type consistency in this project, so it would need to be `unknown[]`.

Reply:
@santam85 I think I misunderstood your previous comment. Are you suggesting to use `Array<any>` in the core repo?

If so, I would not go that way because `any` isn't good for typing, `unknown` is better because it forces a check of the actual type before using it. Using an alias such as `AnyData` would allow to extend this data type later (if needed) without breaking existing integrations (e.g. `type AnyData = AnyArray | AnyObject`) or `type AnyData = (number | string | DataObject)[]`, etc.).

Reply:
Yeah, I meant in the core repo. Sounds good, will adjust accordingly once decided.